### PR TITLE
Initial json

### PR DESCRIPTION
**Pull request by tomsonpl**

### CODE REVIEW

File: src/components/controlListGroup.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const ListGroup = ({ group, name }) => {
+    const listItems = group.map(item => {
+        return (
+            <ListItem
+                controls={item}
+                key={Math.random()}
+            />
+        )
+    });
```
Comment (wdamianw) [inline src/components/controlListGroup.js]:
```
const listItems = group.map(item => (
    <ListItem
        controls={item}
        key={Math.random()}
    />
));
```

File: src/App.js [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
                 });
             })
             .catch((err)=> console.log('err', err))
     }
 
+
     onChange = (newValue) => {
-        this.setState(() => {
-            return { json: newValue }
+        this.setState((prevState) => {
```
Comment (wdamianw) [inline src/App.js]: Unused property.

File: src/components/controlListItem.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+const ListItem = ({ controls }) => {
+    let listItems;
+        if (controls.length > 1) {
```
Comment (wdamianw) [inline src/components/controlListItem.js]: Format the code please.

File: src/components/controlListItem.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            listItems = controls.map(item => {
```
Comment (wdamianw) [inline src/components/controlListItem.js]: Please apply the above rule here as well.

File: src/components/controlsList.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import ListGroup from './controlListGroup';
+
+const List = ({ data }) => {
+    const listItems = data.FormItems.map(item => {
+        return (
```
Comment (wdamianw) [inline src/components/controlsList.js]: As above.

File: src/components/hoc/listHOC.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import List from '../controlsList';
+
+const ListHOC = ({ data }) => {
```
Comment (wdamianw) [inline src/components/hoc/listHOC.js]: As above.

Comment (wdamianw) [reply src/components/hoc/listHOC.js]: `const ListHOC = ({ data }) => (data ? <List data={data} /> : <div>Loading...</div>);`
Doesn't it work?

Comment (tomsonpl) [reply src/components/hoc/listHOC.js]: works, without ()

**Pull request merged**